[quant] quantized reflection_pad1d

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#36450 [quant] quantized reflection_pad1d**
* #36449 [quant] Generalizing _calculate_dynamic_qparams in quantized test

Differential Revision: [D20984967](https://our.internmc.facebook.com/intern/diff/D20984967)

Commit:
[quant] quantized reflection_pad1d

[ghstack-poisoned]

jerryzh168 commented on aten/src/ATen/native/ReflectionPad.cpp:
```diff
@@ -479,7 +499,14 @@ Tensor& reflection_pad1d_out_cpu(
-  auto output = at::empty({0}, input.options());
+  Tensor output;
+  if (input.is_quantized()) {
+    output = at::_empty_affine_quantized({0}, input.options(),
```
this is assuming input is per tensor quantized, add a check for this?

z-a-f replied:
Good point -- I was only doing per_tensor. Will add an explicit check for it

Commit:
Update on "[quant] quantized reflection_pad1d"


Differential Revision: [D20984967](https://our.internmc.facebook.com/intern/diff/D20984967)

[ghstack-poisoned]

Commit:
Update on "[quant] quantized reflection_pad1d"


Differential Revision: [D20984967](https://our.internmc.facebook.com/intern/diff/D20984967)

[ghstack-poisoned]

Commit:
Update on "[quant] quantized reflection_pad1d"


Differential Revision: [D20984967](https://our.internmc.facebook.com/intern/diff/D20984967)

[ghstack-poisoned]

Commit:
Update on "[quant] quantized reflection_pad1d"


Differential Revision: [D20984967](https://our.internmc.facebook.com/intern/diff/D20984967)

[ghstack-poisoned]

Commit:
Update on "[quant] quantized reflection_pad1d"


Differential Revision: [D20984967](https://our.internmc.facebook.com/intern/diff/D20984967)

[ghstack-poisoned]

vkuzo reviewed (approved):
lgtm

Commit:
Update on "[quant] quantized reflection_pad1d"


Differential Revision: [D20984967](https://our.internmc.facebook.com/intern/diff/D20984967)

[ghstack-poisoned]

Commit:
Update on "[quant] quantized reflection_pad1d"


Differential Revision: [D20984967](https://our.internmc.facebook.com/intern/diff/D20984967)

[ghstack-poisoned]

Commit:
Update on "[quant] quantized reflection_pad1d"


Differential Revision: [D20984967](https://our.internmc.facebook.com/intern/diff/D20984967)

[ghstack-poisoned]

Commit:
Update on "[quant] quantized reflection_pad1d"


Differential Revision: [D20984967](https://our.internmc.facebook.com/intern/diff/D20984967)

[ghstack-poisoned]

Commit:
Update on "[quant] quantized reflection_pad1d"


Differential Revision: [D20984967](https://our.internmc.facebook.com/intern/diff/D20984967)

Commit:
Update on "[quant] quantized reflection_pad1d"


Differential Revision: [D20984967](https://our.internmc.facebook.com/intern/diff/D20984967)
